src/caja-location-bar.c

```diff
@@ -263,8 +267,17 @@ style_set_handler (GtkWidget *widget, GtkStyle *previous_style)
     pango_layout_get_pixel_size (layout, &width2, NULL);
     width = MAX (width, width2);
 
+#if GTK_CHECK_VERSION (3, 0, 0)
+    gtk_widget_get_margin_start (widget);
+    gtk_widget_get_margin_end (widget);
+    margin_top = gtk_widget_get_margin_top (widget);
+    margin_bottom = gtk_widget_get_margin_bottom (widget);
+    xpad = margin_start + margin_end;
+    ypad = margin_top + margin_bottom;
```

Comment:
Minor thing: ```ypad```, ```margin_top```, ```margin_bottom``` are only assigned here and never used elsewhere - so they can be safely removed.

Major thing: ```margin_start``` and ```margin_end``` are not initialized before use; did you mean something like this instead?

```C
margin_start = gtk_widget_get_margin_start (widget);
margin_end = gtk_widget_get_margin_end (widget);
```